vumi/middleware/base.py

```diff
@@ -29,12 +37,16 @@ class BaseMiddleware(object):
 
     If you are subclassing this class, you should not override
     :meth:`__init__`. Custom setup should be done in
-    :meth:`setup_middleware` instead.
+    :meth:`setup_middleware` instead. The config class can be overidden by
+    replacing the ``config_class`` class variable.
     """
+    config_class = BaseMiddlewareConfig
 
     def __init__(self, name, config, worker):
         self.name = name
-        self.config = config
+        self.config = self.config_class(config)
```

Comment:
I think this should follow the pattern applications and transports use, i.e. ``self.config = self.CONFIG_CLASS(self.config, static=True)``. This would mean all middleware config is static, which I think is fine (having it depend on the message would just be complicated).